pkg/envoytest/harness.go

```diff
@@ -43,7 +38,71 @@ func GetLoopbackIp(ctx context.Context) (string, error) {
 	return strings.TrimSpace(string(bs)), nil
 }
 
-var cidCounter int64
+func getOSSHome(ctx context.Context) (string, error) {
+	dat, err := dexec.CommandContext(ctx, "git", "rev-parse", "--show-toplevel").Output()
+	if err != nil {
+		return "", err
+	}
+	return strings.TrimSpace(string(dat)), nil
+}
+
+func getLocalEnvoyImage(ctx context.Context) (string, error) {
+	// TODO(lukeshu): Consider unifying GetLocalEnvoyImage() with
+	// agent_test.go:needsDockerBuilds().
+	if env := os.Getenv("ENVOY_DOCKER_TAG"); env != "" { // Same env-var as tests/utils.py:assert_valid_envoy_config()
+		return env, nil
+	}
+
+	ossHome, err := getOSSHome(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	if err := dexec.CommandContext(ctx, "make", "-C", ossHome, "docker/base-envoy.docker.tag.local").Run(); err != nil {
```

Comment:
I'm not in love with this because it puts a hard dependency between how we name stuff in our `make` files and requires us to remember that this exist. CI should ultimately catch it but its not great. However, I don't have a better idea at this time.